WordPress/Classes/ViewRelated/NUX/SigninWPComSyncHandler.swift

```diff
@@ -1,108 +0,0 @@
-        let accountFacade = AccountServiceFacade()
-        let account = accountFacade.createOrUpdateWordPressComAccount(withUsername: username, authToken: authToken)
-
-        // Create reusable success and failure blocks to share between service calls.
-        let successBlock = { [weak self] in
-            accountFacade.updateUserDetails(for: account, success: { [weak self] in
-                self?.handleSyncSuccess(for: account, requiredMultifactor: requiredMultifactor)
-
-                }, failure: { [weak self] (error: Error?) in
-                    self?.handleSyncFailure(error as NSError?)
-            })
-        }
-
-        let failureBlock: (Error?) -> Void = { [weak self] (error: Error?) in
-            self?.handleSyncFailure(error as NSError?)
-        }
-
-        let accountService = AccountService(managedObjectContext: ContextManager.sharedInstance().mainContext)
-        if isJetpackLogin && !accountService.isDefaultWordPressComAccount(account) {
-            let blogService = BlogService(managedObjectContext: ContextManager.sharedInstance().mainContext)
-            blogService.associateSyncedBlogs(toJetpackAccount: account, success: successBlock, failure: failureBlock)
-
-        } else {
-            accountFacade.setDefaultWordPressComAccount(account)
-            BlogSyncFacade().syncBlogs(for: account, success: successBlock, failure: failureBlock)
-        }
-    }
-
-
-    /// Cleans up the view after a successful sync and dismisses the NUX controller.
-    /// - Parameter requiredMultifactor: Whether a multifactor code was required while authenticating.
-    ///
-    /// - Parameters:
-    ///
-    func handleSyncSuccess(for account: WPAccount, requiredMultifactor: Bool) {
-        configureStatusLabel("")
-        configureViewLoading(false)
-
-        // HACK: An alternative notification to LoginFinished.
-        // Observe this instead of `WPSigninDidFinishNotification`
-        // for Jetpack logins.  When WPTabViewController no longer destroy's
-        // and rebuilds the view hierarchy this alternate notification can be
-        // removed.
-        let notification = isJetpackLogin ? .wordpressLoginFinishedJetpackLogin : Foundation.Notification.Name(rawValue: WordPressAuthenticator.WPSigninDidFinishNotification)
-        NotificationCenter.default.post(name: notification, object: account)
-
-        dismiss()
-
-        let properties = [
-            "multifactor": requiredMultifactor ? true.description : false.description,
-            "dotcom_user": true.description
-        ]
-
-        WordPressAuthenticator.post(event: .signedIn(properties: properties))
-    }
-
-
-    /// Handles an error while syncing account and blog information for the
-    /// authenticated user.
-    ///
-    func handleSyncFailure(_ error: NSError?) {
-        configureStatusLabel("")
```

Comment:
I don't see `configureStatusLabel("")` being called on failure, unless I've missed it?

Do we need it?

Reply:
The trick on this one is: 8 lines below we're calling `dismiss`, so no matter what we display, goes away in a split second. (So!!) i've opted for simplifying, and now it's gone.

Sir you've got eagle eyes!! nice find!!